File: gluecodium/src/test/resources/smoke/external_types/output/android/com/example/smoke/Enums.java
/*
 *
 */
package com.example.smoke;
import android.support.annotation.NonNull;
import com.example.NativeBase;
public final class Enums extends NativeBase {
    public enum ExternalEnum {
        FOO_VALUE(0),
        BAR_VALUE(1);
        public final int value;
        ExternalEnum(final int value) {
            this.value = value;
        }
    }
    public enum VeryExternalEnum {
        FOO(0),
        BAR(1);
        public final int value;
        VeryExternalEnum(final int value) {
            this.value = value;
        }
    }
    /**
     * For internal use only.
     * @hidden
     */
    protected Enums(final long nativeHandle, final Object dummy) {
        super(nativeHandle, new Disposer() {
            @Override
            public void disposeNative(long handle) {
                disposeNativeHandle(handle);
            }
        });
    }
    private static native void disposeNativeHandle(long nativeHandle);
    public static native void methodWithExternalEnum(@NonNull final Enums.ExternalEnum input);
}

File: gluecodium/src/test/resources/smoke/external_types/output/android-kotlin/com/example/smoke/Enums.kt
/*

 *
 */

@file:JvmName("EnumsExtensions")

package com.example.smoke

import com.example.NativeBase

class Enums : NativeBase {

    enum class ExternalEnum(@JvmField val value: Int) {
        FOO_VALUE(0),
        BAR_VALUE(1);
    }
    enum class VeryExternalEnum(@JvmField val value: Int) {
        FOO(0),
        BAR(1);
    }


    /**
     * For internal use only.
     * @suppress
     * @param nativeHandle The handle to resources on C++ side.
     * @param tag Tag used by callers to avoid overload resolution problems.
     */
    protected constructor(nativeHandle: Long, @Suppress("UNUSED_PARAMETER") tag: Any?)
        : super(nativeHandle, { disposeNativeHandle(it) }) {}







    companion object {
        @JvmStatic private external fun disposeNativeHandle(nativeHandle: Long)


        @JvmStatic external fun methodWithExternalEnum(input: Enums.ExternalEnum) : Unit
    }
}

